onnxruntime/core/providers/cpu/cpu_execution_provider.cc

```diff
@@ -216,7 +216,7 @@ class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain,
 class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 8, MLFloat16, Expand);
 class ONNX_OPERATOR_VERSIONED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 8, 8, Scan);
 class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 1, If);
-class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 1, Loop);
+class ONNX_OPERATOR_VERSIONED_KERNEL_CLASS_NAME(kCpuExecutionProvider, kOnnxDomain, 1, 11, Loop);
 
```

Comment:
1, 10

Reply:
To do so is inconsistent with the existing approach isn't it? Inconsistencies create confusion.  

Currently I believe we have an entry for each opset range a kernel definition was valid for. When the definition changes we usually create a versioned entry for the previous valid range, and add an unversioned entry starting at the latest opset. AFAIK every currently valid kernel has an unversioned entry with the latest sinceVersion for the kernel.

If we want to change to only having versioned entries I think all declarations should be updated to that style to be consistent, and we need to make sure that doesn't break when opset 12 is created and a kernel definition hasn't changed (i.e. is the unversioned entry required or not).

Reply:
> Per my understanding, I feel like it is not _necessary_ to have an entry for 1, 10 and have an unversioned entry for 11 honestly. Is there a specific reason why that is required ?

My understanding is that a separate loop-11 entry is required. The graph resolution logic will explicitly look for a loop op with a start-version of 11 (i.e. the version range is less important than the start-version of an op). If the onnx spec has a new version of loop starting at 11, ORT logic requires an op with the same start-version.

So, besides the code inconsistency mentioned above, i believe models stamped as opset 11 will fail to load unless we register a loop op for version 11.

Reply:
@jignparm - That is strange - because my test seems to pass. Initially it couldn't resolve a kernel for Loop-11. But after I included it as the *end version*, it could resolve it. So not sure, if it requires one with the same *start version*. It does seem meaningful to be consistent though as @skottmckay points out and hence I will register one with start version as 11.